features/step_definitions/collection.js

```diff
@@ -0,0 +1,267 @@
+            .then((response) => {
+                Assert.oneOf(response.statusCode, [409, 201]);
+
+                if (response.statusCode === 201) {
+                    this.firstCollectionId = response.body.id;
+                } else {
+                    const str = response.body.message;
+
+                    this.secondCollectionId = str.split(': ')[1];
+                }
+            });
+    });
+
+    When(/^they fetch all their collections$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.statusCode, 200);
+        });
+    });
+
+    Then(/^they can see those collections$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.body.length, 2);
+            Assert.strictEqual(response.body[0].name, 'anotherCollection');
+            Assert.strictEqual(response.body[1].name, 'myCollection');
+        });
+    });
+
+    When(/^they delete that collection$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections/${this.firstCollectionId}`,
+            method: 'DELETE',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.oneOf(response.statusCode, [404, 204]);
+        });
+    });
+
+    Then(/^that collection no longer exists$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections/${this.firstCollectionId}`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.statusCode, 404);
+        });
+    });
+
+    Then(/^that pipeline still exists$/, function step() {
+        return this.ensurePipelineExists({ repoName: this.repoName });
+    });
+
+    Then(/^they receive an error regarding unique collections$/, function step() {
+        return createCollection.call(this, { name: 'myCollection' })
+            .then((response) => {
+                Assert.strictEqual(response.statusCode, 400);
+            });
+    });
+
+    After('@collections', function hook() {
+        // Delete the collections created in the functional tests if they exist
+        return Promise.all([
+            deleteCollection.call(this, this.firstCollectionId),
+            deleteCollection.call(this, this.secondCollectionId)
+        ]);
```

Comment:
The cleanup needs to be done in the `After` because for each scenario a new world gets created so the `firstCollectionId` and `secondCollectionId` variables don't exist in the `Before` for the following scenario.

Reply:
Curious: what happens in the event that this `After` fails? Does cucumber bother to execute the remaining scenarios?

Reply:
Just tested it out locally and yes it executes all the scenarios even if the `After` fails.